### PR TITLE
RUN-1950: Grails 6 Upgrade

### PR DESCRIPTION
**Pull request by ahormazabal**
Upgrade to Grails 6 plus associated library version upgrades
Includes upgrade to snakeyaml 2

### CODE REVIEW

File: core/src/test/java/com/dtolabs/rundeck/core/common/TestNodesYamlGenerator.java [hunk 1]
```diff
@@ -228,7 +228,7 @@ public void testGenerate() throws Exception {
     }
 
     private Map parseYamlMap(String outputString) {
-        Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()));
+        Yaml yaml = new Yaml(new LoaderOptions());
```
Comment (gschueler) [inline core/src/test/java/com/dtolabs/rundeck/core/common/TestNodesYamlGenerator.java:231]: not sure why this change was clobbered, I think since we already merged forward compatible changes for snakeyaml2
```suggestion
        Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()));
```

File: core/src/main/java/com/dtolabs/rundeck/core/common/NodesYamlParser.java [hunk 2]
```diff
@@ -69,7 +69,7 @@ public void parse() throws NodeFileParserException {
             throw new NullPointerException("file or inputStream was not set");
         }
         final Reader reader;
-        final Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()));
+        final Yaml yaml = new Yaml(new LoaderOptions());
```
Comment (gschueler) [inline core/src/main/java/com/dtolabs/rundeck/core/common/NodesYamlParser.java:72]:
```suggestion
        final Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()));
```

File: grails-webhooks/src/main/groovy/webhooks/exporter/WebhooksProjectExporter.groovy [hunk 3]
```diff
@@ -44,7 +45,7 @@ class WebhooksProjectExporter implements ProjectDataExporter {
     @Override
     void export(String project, def zipBuilder, Map<String, String> exportOptions) {
         logger.info("Project Webhook export running")
-        Yaml yaml = new Yaml(new SafeConstructor())
+        Yaml yaml = new Yaml(new LoaderOptions())
```
Comment (gschueler) [inline grails-webhooks/src/main/groovy/webhooks/exporter/WebhooksProjectExporter.groovy:48]:
```suggestion
        Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()))
```

File: grails-webhooks/src/main/groovy/webhooks/importer/WebhooksProjectImporter.groovy [hunk 4]
```diff
@@ -59,7 +60,7 @@ class WebhooksProjectImporter implements ProjectDataImporter {
             return ["Import file ${WebhooksProjectExporter.WEBHOOKS_YAML_FILE} was not found"]
         }
         def errors = []
-        Yaml yaml = new Yaml(new SafeConstructor())
+        Yaml yaml = new Yaml(new LoaderOptions())
```
Comment (gschueler) [inline grails-webhooks/src/main/groovy/webhooks/importer/WebhooksProjectImporter.groovy:63]:
```suggestion
        Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()))
```

File: plugins/stub-plugin/src/main/java/com/dtolabs/rundeck/plugin/stub/StubDataStep.java [hunk 5]
```diff
@@ -140,7 +141,7 @@ public static Properties parseData(final String format, final String data) throw
             }
 
         } else if ("yaml".equals(format)) {
-            Yaml yaml = new Yaml(new SafeConstructor());
+            Yaml yaml = new Yaml(new LoaderOptions());
```
Comment (gschueler) [inline plugins/stub-plugin/src/main/java/com/dtolabs/rundeck/plugin/stub/StubDataStep.java:144]:
```suggestion
            Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()));
```

File: rundeckapp/src/main/groovy/org/rundeck/app/components/JobYAMLFormat.groovy [hunk 6]
```diff
@@ -42,7 +43,7 @@ class JobYAMLFormat implements JobFormat {
     @Override
     @CompileStatic(TypeCheckingMode.SKIP)
     List<Map> decode(final Reader reader) throws JobDefinitionException {
-        Yaml yaml = new Yaml(new SafeConstructor())
+        Yaml yaml = new Yaml(new LoaderOptions())
```
Comment (gschueler) [inline rundeckapp/src/main/groovy/org/rundeck/app/components/JobYAMLFormat.groovy:46]:
```suggestion
        Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()))
```

File: rundeckapp/src/main/groovy/org/rundeck/security/RundeckPluginBlocklist.groovy [hunk 7]
```diff
@@ -32,7 +33,7 @@ class RundeckPluginBlocklist implements PluginBlocklist {
     @CompileDynamic
     private load() {
         if (!loaded) {
-            Yaml yaml = new Yaml(new SafeConstructor());
+            Yaml yaml = new Yaml(new LoaderOptions());
```
Comment (gschueler) [inline rundeckapp/src/main/groovy/org/rundeck/security/RundeckPluginBlocklist.groovy:36]:
```suggestion
            Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()));
```

File: scripts/old/circle-build.sh [hunk 8]
```diff
@@ -20,6 +20,13 @@ main() {
 }
 
 build() {
```
Comment (gschueler) [inline scripts/old/circle-build.sh:22]: can the scripts/old/* be removed?

File: core/src/test/java/com/dtolabs/rundeck/core/common/TestNodesYamlGenerator.java (hunk 1 above)
Comment (ahormazabal) [reply core/src/test/java/com/dtolabs/rundeck/core/common/TestNodesYamlGenerator.java:231]: Safeconstructor was needed in 1.x as a mitigation for the cve. Snakeyaml 2 uses safeconstructor as default so I went for the cleaner option. But in the end both forms are exactly the same.

File: scripts/old/circle-build.sh (hunk 8 above)
Comment (ahormazabal) [reply scripts/old/circle-build.sh:22]: Yes we could. I chose to kept them for a while as reference in case something comes up. I'd recommend keeping them at least until we confirm the nightlies, snapshots and releases work OK and I didn't miss anything in the refactoring.

Comment (gschueler) [reply scripts/old/circle-build.sh:22]: we can refer to the old git versions if necessary, I'd say let's delete them, perhaps in another PR

**Review by gschueler — APPROVED** [review]

**Review by gschueler — APPROVED** [review]
